onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -24,6 +24,13 @@ static constexpr bool IsOnnxDomain(std::string_view domain) {
   return (domain == onnxruntime::kOnnxDomain) || (domain == onnxruntime::kOnnxDomainAlias);
 }
 
+// Returns true if the given tensor shape represents a Scalar value.
+// A scalar has a rank of 0 or has shape of (1,).
```

Comment:
Since this is being used for Q/DQ, checking for 1 elem tensors is done for consistency with CPU EP's handling of Q/DQ:
https://github.com/microsoft/onnxruntime/blob/28c252c77e04ade343fdc816704b6906892a1274/onnxruntime/core/providers/cpu/quantization/quantize_linear.cc#L72

But you're right about the naming. Perhaps we can rename this function to `IsScalarOr1ElementTensor`?

Reply:
`IsScalarOr1ElementVector` is good, maybe slightly more precise since there are other tensor shapes which also have one element.

Reply:
Thanks. Renamed to `IsScalarOr1Element1DTensor` to indicate that we're looking only for rank 0 or shape of (1,). The function returns `false` for other shapes like (1, 1, 1) .